packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useListIdent.ts

```diff
@@ -155,14 +155,18 @@ export default function useListIdent(CodeMirror: any) {
 		// otherwise fallback on the default codemirror behavior
 		if (ranges.length === 1) {
 			const line = cm.getLine(anchor.line);
+			// if cursor on 0th line set previousLine=''
+			const previousLine = anchor.line ? cm.getLine(anchor.line - 1) : '';
 
 			if (markdownUtils.isEmptyListItem(line)) {
 				const tokens = cm.getLineTokens(anchor.line);
 				//  A empty list item with an indent will have whitespace as the first token
 				if (tokens.length > 1 && tokens[0].string.match(/^\s/)) {
 					cm.execCommand('smartListUnindent');
-				} else {
+				} else if (markdownUtils.isListItem(previousLine)) {
 					cm.replaceRange('', { line: anchor.line, ch: 0 }, anchor);
+				} else { // perform normal enter key operation
```

Comment:
The code inside this block is just for the case where a normal enter key operation isn't desired. Take a look at the code after this if statement. It performs the normal enter operation. I would suggest moving the `markdownUtils.isListItem(previousLine)` check next to the `isEmptyListItemCheck` so that the return statement is never reached and the normal enter code will be run.

Reply:
Thanks you for the guidance ; I tried these changes but the command [`cm.execCommand('newlineAndIndentContinueMarkdownList');`](https://github.com/laurent22/joplin/blob/3a60efa15360615e890bd42097068d7d7ad3f1ee/packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useListIdent.ts#L178)
that runs when return is not reached causes the same problem that we're trying to solve through these changes; it deletes an item even if there was no list item before it. 
i.e. `1.  `  is removed on pressing enter.
And behaviour of this command is due to a subfile  of node_modules folder; code can be seen [here](https://codemirror.net/addon/edit/continuelist.js)

Reply:
> cm.execCommand('newlineAndIndentContinueMarkdownList');

![ezgif com-gif-maker](https://user-images.githubusercontent.com/63918341/113253547-04c23500-92e3-11eb-8334-2c3446ebf6e0.gif)
Here's a recording of my console demonstrating how content of line 0 changes from `- `  to `<empty string>` on running `newlineAndIndentContinueMarkdown` command from chromium console.

Reply:
Yes you're right, thank you for making it clear. I'm sorry about being delayed in reviewing this.